.github/workflows/test-ui.yaml

```diff
@@ -10,18 +10,22 @@ on:
       - ".github/actions/build-ui"
       - ".github/workflows/test-ui.yaml"
       - "src/ui/package.json"
-      - "src/ui/package-lock.json"
+      - "src/ui/pnpm-lock.yaml"
   workflow_dispatch:
 
 jobs:
   tests:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
+
+      - name: Compile UI assets
+        uses: ./.github/actions/build-ui
+
       - uses: actions/setup-python@v4
         with:
           python-version: "3.10"
-      - uses: ./.github/actions/build-ui
+          cache: "pip"
       - run: |
           python -m pip install --upgrade pip setuptools wheel
           pip install --upgrade tox virtualenv
```

Comment:
Could we clean this a bit for all actions and instructions?

Ref. action, could we clean a bit?:
```yaml
steps:
- uses: actions/checkout@v3
- uses: actions/setup-python@v4
  with:
    python-version: '3.9'
    cache: 'pip' # caching pip dependencies
- run: pip install -r requirements.txt
```

to something like:

Locally I could do with just for building:

```shell
pip install --upgrade pip build setuptools
python3 -m build
```

And for testing

```shell
pip install --upgrade pip tox
tox -e py39
```

`tox` installs all it needs into an isolated environment anyways.

@tomrtk thoughts?  I've done the cleanest steps myself above and it works.  Yet to try them on Windows that is.

Reply:
Sure, try it:)

Reply:
Seems to work. 👍🏻